datalake/src/main/java/com/sequenceiq/datalake/cm/ClouderaManagerRangerUtil.java

```diff
@@ -0,0 +1,124 @@
+import com.google.common.base.Joiner;
+import com.google.common.base.Joiner.MapJoiner;
+import com.google.common.collect.ImmutableSortedMap;
+import com.google.common.collect.Iterables;
+import com.sequenceiq.cloudbreak.cm.client.retry.ClouderaManagerApiFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.RetryException;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+public class ClouderaManagerRangerUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClouderaManagerRangerUtil.class);
+
+    private static final String RANGER_SERVICE_NAME = "ranger";
+
+    private static final String RANGER_USER_SYNC_ROLE_TYPE = "RANGER_USERSYNC";
+
+    private static final String AZURE_USER_MAPPING = "ranger_usersync_azure_user_mapping";
+
+    private static final String AZURE_GROUP_MAPPING = "ranger_usersync_azure_group_mapping";
+
+    private static final MapJoiner CLOUD_IDENTITY_CONFIG_MAP_JOINER =
+            Joiner.on(";").withKeyValueSeparator("=");
+
+    @Inject
+    private ClouderaManagerProxiedClientFactory clouderaManagerProxiedClientFactory;
+
+    @Inject
+    private ClouderaManagerApiFactory clouderaManagerApiFactory;
+
+    private String getClusterName(ApiClient client) throws ApiException {
+        ClustersResourceApi clustersResource = clouderaManagerApiFactory.getClustersResourceApi(client);
+        ApiClusterList clusterList = clustersResource.readClusters(null, null);
+        return Iterables.getOnlyElement(clusterList.getItems()).getName();
+    }
+
+    private String getRangerUserSyncRoleName(ApiClient client, String clusterName) throws ApiException {
+        RolesResourceApi rolesResourceApi = clouderaManagerApiFactory.getRolesResourceApi(client);
+        List<ApiRole> apiRoleList =  rolesResourceApi.readRoles(clusterName, RANGER_SERVICE_NAME, null, null)
+                .getItems()
+                .stream()
+                .filter(apiRole -> apiRole.getType().equals(RANGER_USER_SYNC_ROLE_TYPE))
+                .collect(Collectors.toList());
+        return Iterables.getOnlyElement(apiRoleList).getName();
+    }
+
+    private ApiConfig newCloudIdentityConfig(String configName, Map<String, String> configValues) {
+        // NOTE: We sort the configs first. This isn't really necessary but is more consistent.
+        ImmutableSortedMap<String, String> configValuesSorted = ImmutableSortedMap.copyOf(configValues);
+        ApiConfig config = new ApiConfig();
+        config.setName(configName);
+        config.setValue(CLOUD_IDENTITY_CONFIG_MAP_JOINER.join(configValuesSorted));
+        return config;
+    }
+
+    @Retryable(value = RetryException.class, maxAttempts = 5, backoff = @Backoff(delay = 500))
```

Comment:
why is this the only API that needs retrying?

Reply:
Refresh specifically might need retrying because another refresh is occurring. Though I've found that there is automatic retry of failed api calls, so I don' think a retry here is necessary anymore. I've removed it.